core/templates/dev/head/pages/exploration_player/ExtractImageFilenamesFromStateService.js

```diff
@@ -0,0 +1,112 @@
+ */
+
+oppia.factory('ExtractImageFilenamesFromStateService', ['HtmlEscaperService',
+  function(HtmlEscaperService) {
+    var INTERACTION_TYPE_MULTIPLE_CHOICE = 'MultipleChoiceInput';
+    var INTERACTION_TYPE_ITEM_SELECTION = 'ItemSelectionInput';
+    var INTERACTION_TYPE_IMAGE_CLICK_INPUT = 'ImageClickInput';
+
+    var filenamesInState = [];
+    var _getAllHtmlOfState = function(state) {
+      var _allHtmlInTheState = [];
+      // The order of the imagenames extracted is same as they appea in a state
+      // . The images should be preloaded in the following order --- content,
+      // customizationArgs of interactions, feedback of outcomes(including
+      // feedback of default outcome if any), hints, solution if any.
+      var stateContentHtml = state.content.getHtml();
+      _allHtmlInTheState.push(stateContentHtml);
+
+      if (state.interaction.id === INTERACTION_TYPE_MULTIPLE_CHOICE ||
+        state.interaction.id === INTERACTION_TYPE_ITEM_SELECTION ) {
+        var customizationArgsHtml = '';
+        state.interaction.customizationArgs.choices.value.forEach(
+          function(value) {
+            customizationArgsHtml = customizationArgsHtml.concat(value);
+          });
+        _allHtmlInTheState.push(customizationArgsHtml);
+      }
+
+      state.interaction.answerGroups.forEach(function(answerGroup) {
+        var answerGroupHtml = answerGroup.outcome.feedback.getHtml();
+        _allHtmlInTheState.push(answerGroupHtml);
+      });
+
+      if (state.interaction.defaultOutcome !== null) {
+        var defaultOutcomeHtml = state.interaction.defaultOutcome.feedback
+          .getHtml();
+        if (defaultOutcomeHtml !== '') {
+          _allHtmlInTheState.push(defaultOutcomeHtml);
+        }
+      }
+
+      state.interaction.hints.forEach(function(hint) {
+        var hintHtml = hint.hintContent.getHtml();
+        _allHtmlInTheState.push(hintHtml);
+      });
+      if (state.interaction.solution !== null) {
+        var solutionHtml = state.interaction.solution.explanation.getHtml();
+        _allHtmlInTheState.push(solutionHtml);
+      }
+      return _allHtmlInTheState;
+    };
+
+    var _extractFilepathValueFromOppiaNonInteractiveImageTag = function(
+        strHtml) {
+      var filenames = [];
+      var dummyElement = document.createElement('div');
+      dummyElement.innerHTML = HtmlEscaperService.escapedStrToUnescapedStr(
+        strHtml);
+
+      var imageTagList = dummyElement.getElementsByTagName(
+        'oppia-noninteractive-image');
+      for (i = 0; i < imageTagList.length; i++) {
+        filenames.push(imageTagList[i].getAttribute('filepath-with-value'));
+      }
+      // The name in the array is stored as '"image.png"'. We need to remove
+      // the inverted commas. We remove the first and the last character from
+      // the string (name).
+      for (var x in filenames) {
+        filenames[x] = filenames[x].slice(1, filenames[x].length - 1);
+      }
```

Comment:
Use `map()` for modifying the filenames.

Reply:
Done!